xmtp_mls/src/groups/sync.rs

```diff
@@ -457,7 +480,7 @@ impl MlsGroup {
             EntityKind::Group,
             msgv1.id,
             |provider| -> Result<(), MessageProcessingError> {
-                self.process_message(openmls_group, provider, msgv1, true)?;
+                await_helper(self.process_message(client, openmls_group, provider, msgv1, true))?;
```

Comment:
looks like we don't need `await_helper` here given that `process_message` is async?

Reply:
`process_message` is async, but `client.process_for_id` requires the callback to be synchronous because it runs in a transaction.

This should all go away once we have asynchronous transaction support

Reply:
nice